nuxeo-management-rest-api/src/main/java/org/nuxeo/rest/management/renditions/RenditionsObject.java

```diff
@@ -0,0 +1,147 @@
+ */
+
+package org.nuxeo.rest.management.renditions;
+
+import java.util.EnumSet;
+import java.util.Set;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.nuxeo.ecm.automation.OperationException;
+import org.nuxeo.ecm.core.bulk.BulkService;
+import org.nuxeo.ecm.core.bulk.message.BulkCommand;
+import org.nuxeo.ecm.core.bulk.message.BulkStatus;
+import org.nuxeo.ecm.core.bulk.message.BulkStatus.State;
+import org.nuxeo.ecm.platform.picture.recompute.RecomputeViewsAction;
+import org.nuxeo.ecm.platform.thumbnail.action.RecomputeThumbnailsAction;
+import org.nuxeo.ecm.webengine.model.WebObject;
+import org.nuxeo.ecm.webengine.model.impl.DefaultObject;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.kv.KeyValueService;
+import org.nuxeo.runtime.kv.KeyValueStore;
+
+/**
+ * @since 11.1
+ */
+@WebObject(type = "renditions")
+public class RenditionsObject extends DefaultObject {
+
+    protected Set<State> notAtWork = EnumSet.of(State.UNKNOWN, State.ABORTED, State.COMPLETED);
+
+    protected Lock picturesLock = new ReentrantLock();
+
+    protected Lock thumbnailsLock = new ReentrantLock();
+
+    public static final String RECOMPUTE_VIEWS_ACTION_COMMAND_ID = RecomputeViewsAction.ACTION_NAME + "_commandId";
+
+    public static final String RECOMPUTE_THUMBNAILS_ACTION_COMMAND_ID = RecomputeThumbnailsAction.ACTION_NAME
+            + "_commandId";
+
+    public static final String THUMBNAILS_DEFAULT_QUERY = "SELECT * FROM Document WHERE ecm:mixinType = 'Thumbnail' AND thumb:thumbnail/data IS NULL AND ecm:isVersion = 0 AND ecm:isProxy = 0 AND ecm:isTrashed = 0";
+
+    public static final String PICTURES_DEFAULT_QUERY = "SELECT * FROM Document WHERE ecm:mixinType = 'Picture' AND picture:views/*/title IS NULL";
+
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("pictures/recompute")
+    public BulkStatus doPostPictures() throws OperationException {
+        BulkService bulkService = Framework.getService(BulkService.class);
+        String commandId = getCommandId(RECOMPUTE_VIEWS_ACTION_COMMAND_ID);
+        if (picturesLock.tryLock()) {
+            try {
+                if (needToRun(bulkService, commandId)) {
+                    commandId = bulkService.submit(new BulkCommand.Builder(RecomputeViewsAction.ACTION_NAME,
+                            PICTURES_DEFAULT_QUERY).user(ctx.getPrincipal().getName())
+                                                   .param(RecomputeViewsAction.PARAM_XPATH, "file:content")
+                                                   .build());
+                    putCommandId(RECOMPUTE_VIEWS_ACTION_COMMAND_ID, commandId);
+                }
+            } finally {
+                picturesLock.unlock();
+            }
+        }
+        return bulkService.getStatus(commandId);
```

Comment:
There're cases which seem strange to me:
- if we have a lock (so on the same node), we return a bulk status whose state is `UNKNOWN`, `SCHEDULED`, `SCROLLING_RUNNING` or `RUNNING`. While `SCHEDULED` can make sense, I found the others confusing regarding the action of the endpoint, should we return a `409` or anything else?
- if command doesn't need to be run, we can get the same state, here what is confusing is that I get a status of another action and not the one I just submitted, should we return a `409` or any other status stating that we see status of another action or anything else?

Reply:
Now in any case where the action is not run by the post, people will get a 409.